spring-kafka/src/main/java/org/springframework/kafka/annotation/DltHandlerMethod.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.kafka.support.EndpointHandlerMethod;
+
+/**
+ * Annotation to configure the {@link org.springframework.kafka.support.EndpointHandlerMethod} for handling the DLT,
+ * as part of {@link RetryableTopic} annotation configuration.
+ * This is equivalent to the builder way of {@link org.springframework.kafka.retrytopic.RetryTopicConfigurer}
+ * configuration via {@link org.springframework.kafka.retrytopic.RetryTopicConfigurationBuilder#dltHandlerMethod(EndpointHandlerMethod)}.
+ *
+ * @author Joo Hyuk Kim
+ *
+ * @since 3.2
+ *
+ * @see org.springframework.kafka.retrytopic.RetryTopicConfigurationBuilder#dltHandlerMethod(String, String)
+ *
+ */
+@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface DltHandlerMethod {
```

Comment:
Well, thanks for looking into this!

I wonder if, according to the annotation-driven model, it would be more natural to have something like this instead:
```
@DltHandler
public void handleFailed(Object message) {
}
```
in the same class where we have that `@RetryableTopic`.

See similar model with `@Retryable` and `@Recover` in Spring Retry.

Reply:
I guess it can be subjective as to what natural is 🤔.

Reply:
> See similar model with `@Retryable` and `@Recover` in Spring Retry.

Thinking now, I wonder how Spring `@Retryable` and `@Recover` are implemented using builder, whether they are also built as one, like `RetryConfiguration` or something.

Reply:
But still, maintainer's design perspective also counts alot, so lemme know what you think, @artembilan! ✌🏼✌🏼

Reply:
Hmmm. yes I said the `RetryTopicConfiguration` is global. At the same time, I also believe that `@RetryableTopic` being for specific service is of capability/choice, not as restriction? That much of freedom feels reasonable knowing that DLT can already be configured via `RetryTopicConfigurationBuilder`.

And as per encapsulation of DLT, we have `RetryTopicConfigurationBuilder` which already allows configuration of DLT, meaning retry and DLT's can be associated?

Reply:
Well, again: with an annotation we already have a scope of the method where it is declared, so comparing `RetryTopicConfigurationBuilder` and annotation configuration is not really one-to-one.
Therefore, if we go the way of annotation DLT, then I'd prefer to have it in a scope where its `@RetryableTopic` counterpart is declared.
Plus we already have a positive experience with the patter we have in Spring Retry.
Or even see here `@KafkaHandler` and its `isDefault()` attribute.
Another example is a `@RestController` with its `@..Mapping` and `@ExceptionHandler`.

So, the general point is: if we have a service and give some its method a specific `@KafkaListener` aspect, the other related aspects must be declared in this service as well.

Reply:
Thank you for the insightful explanation🙏🏽🙏🏽 

Took me a while to realize the differences between the annotation and builder way, and how it all fits overall design of Spring libraries.

It also makes sense to consider the positive feedback for the current state implementation👍🏼. Thanks again @artembilan!